### PR TITLE
Use Markdown for readmes

### PR DESCRIPTION
**Pull request by w8jcik**
## Summary

- All `README` and `README.txt` renamed (`git mv`) to `README.md`.
- Changed format of headers to `#`, `##` and `###`.
- Code sections fenced with `` ``` ``
- Fragments of code in sentences marked with `` ` ``
- Changed format of lists.
- Simplified indentation.
- Replaced `NOTE:` with `**Note**:`, same with `NOTES:`, `WARNING:` and similar.
- Replaced quotes with italics and `*` with emphasis.
- Changed titles from `README` (many of them was like this) to something more meaningful.
- Added some tables.
- Removed some typos and inconsistencies.
- Rewrapped at column `80`.

## Impact

- Markdown improves readability. Compare this https://github.com/apache/incubator-nuttx-apps/blob/master/nshlib/README.txt to this https://github.com/w8jcik/incubator-nuttx-apps/blob/master/nshlib/README.md
- Use standard text format for READMEs. Most of the developers are more familiar with Markdown than the previous format, so it is easier to adapt. Markdown is also simpler.
- It is parsable. Previous format wasn't.
- Part of a bigger effort on improving accessibility of NuttX documentation.

## Testing

- You can clone this fork and look at the readmes in your editor.  For example:
  - `git clone https://github.com/w8jcik/incubator-nuttx-apps.git`
  - Open folder in VSCode.
  - Open terminal and run `find . -name README.md`.
  - Ctrl+Click on list of files in the terminal to open them.
  - Install `bierner.markdown-preview-github-styles` extension to see a preview on the side.
  - Without extension some syntax highlighting should be already present in files.
- You can see the files in GitHub
  - [examples/bastest/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/examples/bastest/README.md)
  - [examples/camera/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/examples/camera/README.md)
  - [examples/flash_test/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/examples/flash_test/README.md)
  - [examples/flowc/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/examples/flowc/README.md)
  - [examples/json/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/examples/json/README.md)
  - [examples/pdcurses/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/examples/pdcurses/README.md)
  - [examples/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/examples/README.md)
  - [examples/tcpblaster/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/examples/tcpblaster/README.md)
  - [examples/telnetd/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/examples/telnetd/README.md)
  - [fsutils/inifile/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/fsutils/inifile/README.md)
  - [gpsutils/minmea/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/gpsutils/minmea/README.md)
  - [gpsutils/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/gpsutils/README.md)
  - [graphics/lvgl/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/graphics/lvgl/README.md)
  - [graphics/nxwidgets/Doxygen/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/graphics/nxwidgets/Doxygen/README.md)
  - [graphics/nxwidgets/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/graphics/nxwidgets/README.md)
  - [graphics/nxwidgets/UnitTests/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/graphics/nxwidgets/UnitTests/README.md)
  - [graphics/nxwm/Doxygen/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/graphics/nxwm/Doxygen/README.md)
  - [graphics/nxwm/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/graphics/nxwm/README.md)
  - [graphics/pdcurs34/pdcurses/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/graphics/pdcurs34/pdcurses/README.md)
  - [graphics/pdcurs34/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/graphics/pdcurs34/README.md)
  - [graphics/tiff/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/graphics/tiff/README.md)
  - [graphics/twm4nx/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/graphics/twm4nx/README.md)
  - [industry/abnt_codi/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/industry/abnt_codi/README.md)
  - [interpreters/bas/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/interpreters/bas/README.md)
  - [interpreters/ficl/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/interpreters/ficl/README.md)
  - [interpreters/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/interpreters/README.md)
  - [modbus/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/modbus/README.md)
  - [netutils/discover/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/netutils/discover/README.md)
  - [netutils/ftpc/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/netutils/ftpc/README.md)
  - [netutils/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/netutils/README.md)
  - [netutils/telnetd/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/netutils/telnetd/README.md)
  - [nshlib/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/nshlib/README.md)
  - [README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/README.md)
  - [system/cdcacm/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/system/cdcacm/README.md)
  - [system/cfgdata/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/system/cfgdata/README.md)
  - [system/composite/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/system/composite/README.md)
  - [system/flash_eraseall/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/system/flash_eraseall/README.md)
  - [system/i2c/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/system/i2c/README.md)
  - [system/nsh/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/system/nsh/README.md)
  - [system/nxplayer/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/system/nxplayer/README.md)
  - [system/psmq/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/system/psmq/README.md)
  - [system/spi/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/system/spi/README.md)
  - [system/termcurses/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/system/termcurses/README.md)
  - [system/usbmsc/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/system/usbmsc/README.md)
  - [system/zmodem/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/system/zmodem/README.md)
  - [testing/cxxtest/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/testing/cxxtest/README.md)
  - [testing/fstest/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/testing/fstest/README.md)
  - [testing/nxffs/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/testing/nxffs/README.md)
  - [testing/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/testing/README.md)
  - [testing/smart/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/testing/smart/README.md)
  - [testing/smart_test/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/testing/smart_test/README.md)
  - [tools/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/tools/README.md)
  - [wireless/bluetooth/btsak/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/wireless/bluetooth/btsak/README.md)
  - [wireless/ieee802154/i8sak/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/wireless/ieee802154/i8sak/README.md)
  - [wireless/wapi/README.md](https://github.com/w8jcik/incubator-nuttx-apps/blob/master/wireless/wapi/README.md)

## Questions
- Is wrap at column 80 fine?
- What should be format of headers. Is it _Other Network Applications_ or _Other network applications_. Previously it was not completely consistent. I find the second form more readable and I would use it, but maybe I am not good with English.
- Can we split group readmes, like this one https://github.com/w8jcik/incubator-nuttx-apps/blob/master/examples/README.md into smaller readmes in each subfolder?
- Do you see some other style violations?

## Notes
- Yes I will squash it when done.
- The GitHub review seems to be a bit broken. Despite me using `git mv` to rename the files
  ```
  find . -name README -exec git mv "{}" "{}.md" \;
  find . -name README.txt -execdir sh -c 'git mv $(printf "%s\n" "${0%.*}").txt $(printf "%s\n" "${0%.*}").md' {} ';'
  ```
  and commiting afterwards, GitHub shows that some files were deleted and created. We could do separate MR just for renaming. Or maybe it is just GitHub that shows it wrongly. It needs to be done correctly to preserve history.
- The PR consists of 101 changed files. Maybe we could split the change into few smaller PRs. For example we could start with `examples` folder and review it first.
- I will do one more check if everything is consintent and nothing is lost, before merging, but for now I would like to see if you are fine with this change and is it going in the right direction. Also we still didn't have a vote on it.

### CODE REVIEW

**Comment by v01d** [comment]
great work! I will have a look at the changes

File: README.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+The default application directory used by the NuttX build should be named
+`apps/` (or `apps-x.y.z/` where `x.y.z` is the NuttX version number). This `apps/`
+directory should appear in the directory tree at the same level as the
+NuttX directory. Like:
+
+```
+ .
+ |- nuttx
+ |
+ `- apps
```
Comment (v01d) [inline README.md:32]: It is easier and simpler to just prepend these kind of verbatim blocks with four spaces. The backticks are not required (unless you need to escape something I think)

**Comment by v01d** [comment]
Regarding your questions:
- I think 80 column is OK and is consistent with C style
- I'm also not a native english speaker but I believe capitalizations in english should be in the first form
- Splitting big READMEs: I think it is useful to have a listing with brief description of all apps available. However it would seem that for some apps there is indeed a more detailed description which would be better to maintain in its own directory. Anyway, if we can move forward with the new doc system, I'm imagine this list could be built by picking up individual READMEs so no need for a manual list of apps.

File: README.md (hunk 1 above)
Comment (w8jcik) [reply README.md:32]: The indented form doesn't allow selecting language for syntax highlighting. We could use indented for non-highlighted fragments and backticks for highlighted fragments. But I think it is better to stick to one.

Also, as far as I remember I always see only the backtick form being used. Example: https://raw.githubusercontent.com/ARMmbed/littlefs/master/README.md

Just for consistency I would stick to triple backtick everywhere.

**Comment by w8jcik** [comment]
These failed macOS builds don't seem to be valid. I didn't change any code, so it must be a problem with the macOS runners or CI script.

**Comment by w8jcik** [comment]
This is a big PR, difficult to review. To make it easier, I created another PR here https://github.com/apache/incubator-nuttx-apps/pull/339. It is limited to one directory.

**Comment by w8jcik** [comment]
I have one more idea. If some maintainer could create a branch for me called for example `md-readmes`, from `master`, with this executed:

```
find . -name README -exec git mv "{}" "{}.md" \;
find . -name README.txt -execdir sh -c 'git mv $(printf "%s\n" "${0%.*}").txt $(printf "%s\n" "${0%.*}").md' {} ';'
```

then I could do PR to it, and maybe GitHub would show valid diff.

**Comment by xiaoxiang781216** [comment]
@w8jcik you just need reorg the change into two patch:
1.Rename all README to README.md
2.Modify README.md per Markdown style
We can select and review the second patch from 'Commits' tab.

**Comment by w8jcik** [comment]
> We can select and review the second patch carefully from 'Commits' tab.

Great, this one works, thank you :)

Every time I am passing through this files, I find some small inconsistencies. I think I will need to review them once again. For now I would like to know if you guys don't have any major concerns and should I proceed, or maybe we don't need Markdown.

Also how should we tackle review. Is the idea of splitting into smaller PRs a good idea, or should I close the other PR that I created for this purpose?

**Comment by w8jcik** [comment]
I reviewed everything one more time and fixed some consistency issues.

> I can only point to two cases where it would probably be nice to convert the tables to actual tables in abnt_codi and nshlib READMEs.

@v01d They are replaced with tables now.

I consider this PR done and ready to be merged.

**Comment by xiaoxiang781216** [comment]
Looks great, @w8jcik ! @v01d can you merge this change?

**Comment by btashton** [comment]
@xiaoxiang781216 I don't believe he can, but I will.  @w8jcik this work is really appreciated.

**Pull request merged**